lib/_stream_readable.js

```diff
@@ -311,8 +311,7 @@ function chunkInvalid(state, chunk) {
 // 'readable' event will be triggered.
 function needMoreData(state) {
   return !state.ended &&
-         (state.length < state.highWaterMark ||
-          state.length === 0);
+         (state.length < state.highWaterMark);
 }
```

Comment:
Suggestion: this function could be inlined. It only has a single call site.

Reply:
Also needs to be `<=` as above

Reply:
with <= , a test case is failing `AssertionError [ERR_ASSERTION]: true strictEqual false
    at Object.<anonymous> (/Users/vishal/node/test/parallel/test-stream2-objects.js:212:12)`

Reply:
can you remove the parenthesis? They are not needed anymore.

Reply:
can prob also be a one liner as this looks like it's <80 chars

Reply:
yes. I will do that